src/peep/staff.c

```diff
@@ -268,19 +271,33 @@ void game_command_hire_new_staff_member(int* eax, int* ebx, int* ecx, int* edx,
 
 		RCT2_ADDRESS(RCT2_ADDRESS_STAFF_MODE_ARRAY, uint8)[newStaffId] = STAFF_MODE_WALK;
 
-		for (int edi = 0; edi < 0x80; edi++) {
-			int addr = RCT2_ADDRESS_STAFF_PATROL_AREAS + (newStaffId << 9) + edi * 4;
+		for (int i = 0; i < 0x80; ++i) {
+			int addr = RCT2_ADDRESS_STAFF_PATROL_AREAS + (newStaffId << 9) + i * 4;
 			RCT2_GLOBAL(addr, uint32) = 0;
 		}
 	}
 
-	if(staff_type == STAFF_TYPE_HANDYMAN && gConfigGeneral.handymen_mow_default) {
-		int flags = ((newPeep->staff_orders ^ (1 << 3)) << 8) | 1;
-		game_do_command(newPeep->x, flags, newPeep->y, newPeep->sprite_index, GAME_COMMAND_SET_STAFF_ORDER, (int)newPeep, 0);
+	if ((staff_type == STAFF_TYPE_HANDYMAN) && gConfigGeneral.handymen_mow_default) {
+		uint8 new_orders = newPeep->staff_orders | STAFF_ORDERS_MOWING;
+		game_do_command(newPeep->x, ((int)new_orders << 8) | 1, newPeep->y, newPeep->sprite_index, GAME_COMMAND_SET_STAFF_ORDER, (int)newPeep, 0);
```

Comment:
Yeah, that makes sense. What I mean is that `*ebx` in [that expression](https://github.com/OpenRCT2/OpenRCT2/pull/3854/files#diff-f8ffc5a56691b0f5f481bd19d44e680fR318) is the `((int)new_orders << 8) | 1` from  before, where `1` is set in all calls to this function. And so maybe in a new PR I can simplify this too.